lib/ansible/module_utils/network/aci/msc.py

```diff
@@ -0,0 +1,299 @@
+        host=dict(type='str', required=True, aliases=['hostname']),
+        port=dict(type='int', required=False),
+        username=dict(type='str', default='admin'),
+        password=dict(type='str', required=True, no_log=True),
+        output_level=dict(type='str', default='normal', choices=['normal', 'info', 'debug']),
+        timeout=dict(type='int', default=30),
+        use_proxy=dict(type='bool', default=True),
+        use_ssl=dict(type='bool', default=True),
+        validate_certs=dict(type='bool', default=True),
+    )
+
+
+class MSCModule(object):
+
+    def __init__(self, module):
+        self.module = module
+        self.params = module.params
+        self.result = dict(changed=False)
+        self.headers = {'Content-Type': 'text/json'}
+
+        # normal output
+        self.existing = dict()
+
+        # info output
+        self.previous = dict()
+        self.proposed = dict()
+        self.sent = dict()
+
+        # debug output
+        self.filter_string = ''
+        self.method = None
+        self.path = None
+        self.response = None
+        self.status = None
+        self.url = None
+
+        # Ensure protocol is set
+        self.params['protocol'] = 'https' if self.params.get('use_ssl', True) else 'http'
+
+        # Set base_uri
+        if 'port' in self.params and self.params['port'] is not None:
+            self.baseuri = '{protocol}://{host}:{port}/api/v1/'.format(**self.params)
+        else:
+            self.baseuri = '{protocol}://{host}/api/v1/'.format(**self.params)
+
+        if self.module._debug:
+            self.module.warn('Enable debug output because ANSIBLE_DEBUG was set.')
+            self.params['output_level'] = 'debug'
+
+        if self.params['password']:
+            # Perform password-based authentication, log on using password
+            self.login()
+        else:
+            self.module.fail_json(msg="Parameter 'password' is required for authentication")
+
+    def login(self):
+        ''' Log in to MSC '''
+
+        # Perform login request
+        self.url = urljoin(self.baseuri, 'auth/login')
+        payload = {'username': self.params['username'], 'password': self.params['password']}
+        resp, auth = fetch_url(self.module,
+                               self.url,
+                               data=json.dumps(payload),
+                               method='POST',
+                               headers=self.headers,
+                               timeout=self.params['timeout'],
+                               use_proxy=self.params['use_proxy'])
+
+        # Handle MSC response
+        if auth['status'] != 201:
+            self.response = auth['msg']
+            self.status = auth['status']
+            self.fail_json(msg='Authentication failed: {msg}'.format(**auth))
+
+        payload = json.loads(resp.read())
+
+        self.headers['Authorization'] = 'Bearer {token}'.format(**payload)
+
+    def request(self, path, method=None, data=None):
```

Comment:
This was a copy paste and still references Meraki

Reply:
Good catch :-)